consensus/state.go

```diff
@@ -1297,12 +1300,14 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 	stateMachineValidBlock, err := cs.blockExec.ProcessProposal(cs.ProposalBlock, cs.state)
 	if err != nil {
 		cs.Logger.Error("state machine returned an error when trying to process proposal block", "err", err)
+		return
```

Comment:
[Question] Is the addition of this `return` statement intentional?

Reply:
Yes, I don't think we should vote if there is an error here